### PR TITLE
feat(Legend): change click on item behaviour

### PR DESCRIPTION
**Pull request by dej611**
## Summary

Clicking on a legend item has been improved to isolate a series on first click, show all series on second click.
Also, when clicking on a hidden series now will add the selected series into the visible set.

### BREAKING CHANGE
The modifier key has been changed to `CTRL` (or `CMD` on Mac) from `SHIFT`. The `SHIFT` key will no longer have any effect on click.

## Details

A full description of the new behaviour:
> First click isolates selected element, then behave according to context (if single element is selected, restore all, otherwise toggle on off).

### Checklist

- [x] The proper **chart type** label has been added (e.g. `:xy`, `:partition`)
- [x] The proper **feature** labels have been added (e.g. `:interactions`, `:axis`)
- [x] Unit tests have been added or updated to match the most common scenarios

### CODE REVIEW

**Comment by dej611** [comment]
buildkite update screenshots

File: packages/charts/src/state/reducers/interactions.ts [hunk 1]
```diff
@@ -273,19 +273,28 @@ function toggleDeselectedDataSeries(
...
 
-  // todo consider branch simplifications
-  if (negate) {
-    return alreadyDeselected || deselectedDataSeries.length !== legendItemsKeys.length - 1
-      ? legendItems
-          .flatMap(({ seriesIdentifiers }) => seriesIdentifiers)
-          .filter(({ key }) => !actionSeriesKeys.includes(key))
-      : legendItemIds;
-  } else {
+  // when a meta key (CTRL or Mac Cmd ⌘) add or remove the clicked item from the visible list
```
[the remark this answers is not in the document]
Comment (markov00) [reply packages/charts/src/state/reducers/interactions.ts:278]: Please also update that in the PR breaking change description

**Comment by markov00** [comment]
buildkite test this

**Comment by dej611** [comment]
buildkite update screenshots

**Comment by dej611** [comment]
Now all checks are passing @markov00 . It was a wrong casting in the playwright test.

**Comment by dej611** [comment]
buildkite update screenshots

**Comment by dej611** [comment]
buildkite update screenshots

**Comment by dej611** [comment]
buildkite update screenshots

**Comment by dej611** [comment]
buildkite update screenshots

**Comment by dej611** [comment]
buildkite update screenshots

**Comment by dej611** [comment]
All screenshots are now being checked and resolved with no changes from the `main` branch.

**Review by markov00 — APPROVED** [review]

**Pull request merged**